packages/hdwallet-ledger/src/solana.ts

```diff
@@ -0,0 +1,112 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import {
+  AddressLookupTableAccount,
+  ComputeBudgetProgram,
+  PublicKey,
+  SystemProgram,
+  TransactionInstruction,
+  TransactionMessage,
+  VersionedTransaction,
+} from "@solana/web3.js";
+import * as bs58 from "bs58";
+
+import { LedgerTransport } from "./transport";
+import { handleError } from "./utils";
+
+const HARDENED = 0x80000000;
+
+// This ensures we generate a derivation compatible with both hw-app-solana (stripped m/) and Phantom (added hardened change component)
+// Using good ol' core.bip32ToAddressNList here = problems
+// https://help.phantom.com/hc/en-us/articles/12988493966227-What-derivation-paths-does-Phantom-wallet-support#h_01HFBRSYQKC73Z43HPKKW0C1SS
+// https://www.npmjs.com/package/@ledgerhq/hw-app-solana#getaddress
```

Comment:
i.e
- Ledger expects the `m/` prefix to be trimmed
- Phantom derivation path uses the `bip44Change` bip32 path format which is slightly different than ours e.g for account 0: `m/44'/501'/0'/0'` vs. `m/44'/501'/0'/0/0` for us. Note missing `address_index` part, and hardened `change`, which is not hardened in our standard `core.bip32ToAddressNList()` paths.

This ensures that the same seed yields the same address 0 over Phantom and app. Note for account 1+, we also honour the Phantom path, but Phantom has some magic re: accounts derivation so addresses may not match because of autodiscovery, see https://github.com/shapeshift/hdwallet/pull/696/files#r1869379368